### PR TITLE
Replace ng-route with ui-router and add project routes

### PR DESCRIPTION
**Pull request by txels**
https://txels.tpondemand.com/entity/344
This is another one of the cleanups/refactors I wanted to do with the angular code. Good things of this change will be more apparent later as the app grows.

Functionally, this adds routing to individual projects (e.g. /projects/ployst). The end goal is that every single state/page within the app has an individual bookmarkable route.

### CODE REVIEW

File: ployst/ui/static/app.js [hunk 1]
```diff
@@ -20,46 +19,35 @@ var ployst = angular.module('ployst', [
     .constant('Django', {
         URL: URLS
     })
-    .run([
```
Comment (txels) [inline ployst/ui/static/app.js:23]: (moved down to better reflect order of execution, `config` before `run`)

File: ployst/ui/static/navbar/navbar.html [hunk 2]
```diff
@@ -12,16 +12,16 @@
...
                 <li ng-if="!user.is_anonymous" class="dropdown">
-                    <a href="/#" class="dropdown-toggle" data-toggle="dropdown">
+                    <a class="dropdown-toggle" data-toggle="dropdown">
                         <i class="fa fa-cog"></i> {{ user.username }}
                         <b class="caret"></b>
                     </a>
                     <ul class="dropdown-menu">
                         <li>
-                            <a href="{{ URL.HOME }}#/profile"><i class="fa fa-fw fa-user"></i> Profile</a>
+                            <a ui-sref="profile"><i class="fa fa-fw fa-user"></i> Profile</a>
```
Comment (txels) [inline ployst/ui/static/navbar/navbar.html:21]: This is one of the goodies: non-hardcoded URLs.

File: ployst/ui/static/projects/menuProjects.html [hunk 3]
```diff
@@ -9,7 +9,7 @@
         </p>
     </li>
     <li ng-repeat="p in ps.projects" ng-class="{active: p===ps.project}">
-        <a ng-click="ps.selectProject(p)">
+        <a ui-sref="projects({project:p.name})">
```
Comment (txels) [inline ployst/ui/static/projects/menuProjects.html:15]: This is kind of django's reverse URL, in angular ui.router

**Comment by alexcouper** [comment]
FWIW: I like the goal in the description

File: ployst/ui/static/projects/menuProjects.html (hunk 3 above)
Comment (alexcouper) [inline ployst/ui/static/projects/menuProjects.html:15]: pure magic

**Comment by alexcouper** [comment]
Happy that you know what you're doing.

**Comment by txels** [comment]
Happy that it looks like I do :D

**Pull request merged**